src/lib/components/toaster/toasts.ts

```diff
@@ -64,3 +64,25 @@ export function postponeDismissal(id: number, timeout: number): void {
 export function dismissToast(id: number): void {
   toasts.update((all) => all.filter((t) => t.id !== id));
 }
+
+export type doToastOpts<T> = Omit<Omit<Toast, 'id'>, 'content'> & {
+  success: (_: T) => string;
+  failure: (err: unknown) => string;
+};
+
+export async function toastDo<T>(
```

Comment:
Nit: Verb to emit toast is not specific enough. 
Consider: <https://github.com/crcn/programming-terms#mediator> emit methods.

Reply:
Do is about doing the callback, as opposed to just `toast` (which is not even a verb). I don't think any of the verbs listed here are suitable.

Reply:
Even then, we tend to place the verb before the subject.
 
Note: `toast` (and many nouns) may technically be used as a verb, though it is, admittedly, not so obvious depending on the context especially at first glance. I value code being easy on the brains of any programmer to understand at first glace (without further/unneeded/avoidable contextualization), hence the seemingly inane discussion on function naming.

Reply:
> Even then, we tend to place the verb before the subject.

Because `doToast`, which is the common way to word functions, describes a completely different (but a lot more common) functionality. `toastDo` is not.

A longer name for this function would be `toastDoFunc` or `toastDoFn`.



> Note: `toast` (and many nouns) may technically be used as a verb, though it is, admittedly, not so obvious depending on the context.

I'm not aware of `toast` (verb) being used as a verb for spawning a toast notification anywhere. An actual, more common verb would be `popup`.

Reply:
Fixed by making `toast` possibly accept a function.